source/level/LevelGenerator.hx

```diff
@@ -0,0 +1,152 @@
+package level;
+
+import flixel.FlxG;
+
+private class Point {
+	public var x:Int;
+	public var y:Int;
+
+	public function new(x:Int, y:Int) {
+		this.x = x;
+		this.y = y;
+	}
+
+	public function toString():String {
+		return this.x + " " + this.y;
+	}
+}
+
+/**
+ * Contains logic to create random levels.
+ * Call generateLevel() to get a bitmap
+ */
+class LevelGenerator {
+	// Defining static constants
+	private static var MIN_Y_GAP_SIZE(default, never):Int = 3;
+	private static var MIN_X_GAP_SIZE(default, never):Int = 3;
+	private static var MIN_WALL_COUNT(default, never):Int = 5;
+	private static var MAX_WALL_COUNT(default, never):Int = 12;
+	private static var MIN_WALL_SIZE(default, never):Int = 4;
+	private static var MAX_WALL_SIZE(default, never):Int = 12;
+
+	private static var BITMAP_EMPTY_BIT(default, never):Int = 0;
+	private static var BITMAP_WALL_BIT(default, never):Int = 1;
+
+	// Instantiating Instance Variables
+	private var mapWidth:Int;
+	private var mapHeight:Int;
+
+	private var minX:Int;
+	private var minY:Int;
+	private var maxX:Int;
+	private var maxY:Int;
+
+	private var map:Array<Array<Int>>;
+
+	public function new(?mapWidth:Int = 20, ?mapHeight:Int = 20) {
+		this.mapWidth = mapWidth;
+		this.mapHeight = mapHeight;
+	}
+
+	/**
+	 * Creates a randomly generated level in the form of a 2D Array
+	 * @return Array<Array<Int>>	Level map
+	 */
+	public function generateLevel():Array<Array<Int>> {
+		instantiateMap();
+		addBorders();
+		addObstacles();
+
+		return this.map;
+	}
```

Comment:
The way this is set up right now, this level generation object is holding onto a reference to `map` as an instance variable.

It's kinda handy that we can call all of these methods without having to pass it around, but it also strikes me as a bit dangerous - we can write a valid block of code like 
```Haxe
addBorders();
addObstacles();
instantiateMap();
```
and cause an exception to occur. We can see from the naming that we really shouldn't call them like this (always call `instantiate` first, yeah?) but that puts the responsibility on us to stay safe, sorta like having the guard rails taken off of a bridge. The risk shouldn't be very high, but we're allowing a pattern of usage that's not going to be helpful for anyone. If there are any hidden pitfalls (say, `addBorders()` needs to be called before `addObstacles()` it makes it trickier for your teammates to step in and fix up broken functionality.

 This risk would definitely be magnified if we made any of these methods `public` - it'd  be neat if we let the person using `LevelGenerator` choose what sorts of setup steps they want to mix together to create a level, but as-is we'd be requiring them to do things in the appropriate order.

The alternative I'd propose is seeing if there's a good way to make each method (more) self-sufficient. Methods could require that a 2D array be passed in as an argument to the function, and they could return it when they're done.

There are lots of other cool strategies you could use too, but my main point was just that it'd be good to watch out for this sort of pattern in your code - it might make updating your code difficult in ways you didn't expect.